src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  get isAlive() {
+    return this.health > 0;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health <= 0) {
+      Animal.cleanupDeadAnimals();
+    }
+  }
 }
 
+Animal.alive = [];
+Animal.cleanupDeadAnimals = () => {
+  Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      animal instanceof Herbivore
+      && !animal.hidden
+      && animal.isAlive
+    ) {
+      animal.applyDamage(50);
+    }
```

Comment:
The `bite` method correctly checks if the target is an instance of `Herbivore`, is not hidden, and is alive before applying damage. This adheres to the task requirements.